drivers/watchdog/Kconfig.sam

```diff
@@ -6,8 +6,8 @@
 #
 
 menuconfig WDT_SAM
-	bool "Atmel SAM MCU Family Watchdog (WDT) Driver"
-	depends on SOC_FAMILY_SAM
```

Comment:
There should be no need to touch this file anymore since SAM D21 belongs to a different family now.

Reply:
Done.